iModelCore/ECDb/ECDb/ECSql/ECSqlFieldFactory.cpp

```diff
@@ -299,11 +299,15 @@ ECSqlColumnInfo ECSqlFieldFactory::CreateColumnInfoForProperty(ECSqlPrepareConte
 //static
 ECSqlColumnInfo ECSqlFieldFactory::CreateTopLevelColumnInfo(IssueDataSource const& issues, bool isSystemProperty, bool isGeneratedProperty, ECSqlPropertyPath const& propertyPath, ECSqlColumnInfo::RootClass const& rootClass, ECPropertyCP originalProperty, bool isDynamic)
     {
-    BeAssert(propertyPath.Size() > 0);
-    ECPropertyCP ecProperty = propertyPath.GetLeafEntry().GetProperty();
-    BeAssert(ecProperty != nullptr);
     DateTime::Info dateTimeInfo;
     ECStructClassCP structType = nullptr;
+    BeAssert(propertyPath.Size() > 0);
+    ECPropertyCP ecProperty = propertyPath.GetLeafEntry().GetProperty();
+    //BeAssert(ecProperty != nullptr);
+    if (ecProperty == nullptr)
+        {
+        return ECSqlColumnInfo(ECTypeDescriptor(), dateTimeInfo, structType, ecProperty, originalProperty, isSystemProperty, isGeneratedProperty, propertyPath, rootClass, isDynamic);

```

Comment:
So this is not really a fix. The type of descriptor is empty. This should never be empty of any property. For property p2d.x it should be primitive and double.

Without this information we will not be able to serialize this into json.

Reply:
Also as a rule top level property must have ECProperty set not matter what. We create a statement schema just for this purpose where any expr that is not a direct EC Property we just create a runtime property and add to the statement schema.

Reply:
Also add test on itwin.js side with same schema and query. I am confident with your fix it will still crash but at a different place. ConcurrentQuery uses ECSqlRowAdaptor to convert rows to json. It also assumes top level properties will not be null. As its either ECProperty from class or a generated ECProperty but never null.